### DISCUSSION TITLE
SubRip detected as SRecode Template

### DISCUSSION THREAD

**Discussion by MichaelHinrichs** [github-linguist/linguist#6896] [Classification]
https://github.com/search?q=language%3A%22SRecode+Template%22+&type=repositories
The entire first page, and much of the following pages, are subtitles.

**Comment by lildude** [comment]
So there are a few things here:

1. you're looking at the Search results. Search is a completely independent library from Linguist though it feeds off Linguist
2. the results are showing the predominant language detected in the repo, not the repo's language
3. SubRip isn't likely to appear commonly in those results as it is [considered "data" by Linguist](https://github.com/github-linguist/linguist/blob/39fd5e93de98de3434cb3e857c5b1972c418f8c5/lib/linguist/languages.yml#L6967-L6974) so will only appear on the language stats if someone implements an override.

The fact you're seeing SubRip detected as SRecode will be because both languages share the `.srt` extension and all those repos do not contain files that predominantly match the heuristic:

https://github.com/github-linguist/linguist/blob/39fd5e93de98de3434cb3e857c5b1972c418f8c5/lib/linguist/heuristics.yml#L704-L707

... so things are falling through to the classifier and it's  "knowledge" is based entirely on the samples we have.

If you can improve the heuristic, we're open to a PR.
